daemons/dnscrypt-proxy

```diff
@@ -128,3 +138,97 @@ dnscrypt-roundrobin() {
 
 	rm -f $R/share/dnscrypt-resolvers-grepped.csv
 }
+
+#=======================================
+# Code for check DNS availability
+#=======================================
+__search_for_a_dns () {
+    # 
+    sed 1q $R/share/dnscrypt-resolvers.csv > $R/share/dnscrypt-resolvers-grepped.csv
+    grep $servertype $R/share/dnscrypt-resolvers.csv >> $R/share/dnscrypt-resolvers-grepped.csv
+
+    
+    # prints a random line from the resolvers csv file
+    local _rndresolv="$(perl -wne '$a = $_ if rand $. < 1 ; END { print $a }' $R/share/dnscrypt-resolvers-grepped.csv)"
+    
+    # check if alive
+    eval "${execmap[dnscrypt-proxy]} -L $R/share/dnscrypt-resolvers.csv -R ${_rndresolv[(ws:,:)1]} -t 1 -m 1"
+    
+    [[ $? = 0 ]] || return 
+    
+    dnscrypt_name="${_rndresolv[(ws:,:)1]}"
+
+    # Saving on redis the dnscrypt_name
+    print "SET dns-server-selected ${dnscrypt_name}" | redis dynamic
+    
+    rm -f $R/share/dnscrypt-resolvers-grepped.csv
+
+    PID_LIST=$(print "GET dnscrypt-pid-child-list " | redis dynamic | tr '_' ' ')
+
+    LOCK=$(print "GET lock_check_and_select_dns " | redis dynamic )
+
+    # in this moment we can unlock the semaphore if the pendulum spawn another team it don't change our  PID_LIST variable 
+    unlock check_and_select_dns $LOCK
+    kill -9 ${PID_LIST}  
+    
+}
+
+
+__is_dns_reachable() {
+    eval "${execmap[dnscrypt-proxy]} -L $R/share/dnscrypt-resolvers.csv -R $1 -t 1 -m 1"
```

Comment:
-t argument should be reverted to 720 because we have no use in checking whether a cert expires within one minute.

the -m argument should be an option between 1 (in normal mode) and 7 (in debug mode)